### PR TITLE
Auto load ST on startup and version bump

### PR DESCRIPTION
**Pull request by po09i**
## Description
This PR implements the following:
- Bumps FSLeyes version to 1.9.0
- Version 1.8.3 allowed to define layouts in the plugin, this allows to set up the ST plugin within the layout and therefore auto launch the plugin on startup. This has been a feature we have been wanting for a while #60 
- Bumps the Shimming Toolbox version to https://github.com/shimming-toolbox/shimming-toolbox/pull/483
- Removes the scanner constraints default path
- Add quadprog to the list of optimizers
- Add fieldmap and b0shim tab tests

## Linked Issues
#60

### CODE REVIEW

File: fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py [hunk 1]
```diff
@@ -655,15 +673,20 @@ def create_sizer_realtime_shim(self, metadata=None):
...
             panel=self,
             dropdown_metadata=dropdown_opt_metadata,
             label="Optimizer",
-            option_name = 'optimizer-method',
-            list_components= [dropdown_crit, component_reg_factor, self.create_empty_component()],
-            component_to_dropdown_choice=[0, 0, 1],
+            option_name='optimizer-method',
+            list_components=[dropdown_crit, component_reg_factor_lsq,
+                             self.create_empty_component(), component_reg_factor_qp],
+            component_to_dropdown_choice=[0, 0, 1, 2],
```
Comment (4rnaudB) [inline fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py:689]: 0, 0 ? not -1, 0?

File: fsleyes_plugin_shimming_toolbox/tabs/fieldmap_tab.py [hunk 2]
```diff
@@ -108,8 +138,9 @@ def create_fieldmap_sizer(self):
             dropdown_metadata=dropdown_mask_threshold,
             label="Unwrapping region",
             info_text="Masking methods either with a file input or a threshold",
-            option_name='no_arg',
-            list_components=[self.create_empty_component(), self.component_mask, self.component_threshold],
+            option_name='no_arg_roi',
+            list_components=[self.component_save_mask1, self.component_mask, self.component_threshold, self.component_save_mask2],
+            component_to_dropdown_choice=[0, 1, 2, 2],
```
Comment (4rnaudB) [inline fsleyes_plugin_shimming_toolbox/tabs/fieldmap_tab.py:143]: 2,2?

File: test/gui/test_tabs.py [hunk 3]
```diff
@@ -111,7 +45,30 @@ def get_notebook(view):
...
+def get_tab(nb_terminal, tab_instance):
+    """ Returns the tab instance from the ST notebook."""
+    tabs = nb_terminal.GetChildren()
+    output_tab = None
+    for tab in tabs:
+        if isinstance(tab, tab_instance):
+            output_tab = tab
+            break
+
+    return output_tab
```
Comment (4rnaudB) [inline test/gui/test_tabs.py]:
```suggestion
    for tab in tabs:
        if isinstance(tab, tab_instance):
            return tab
```

File: fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py (hunk 1 above)
Comment (po09i) [reply fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py:689]: Yes, this list mentions what component (or dropdown) of `list_compontents` is related to which option of the dropdown

File: fsleyes_plugin_shimming_toolbox/tabs/fieldmap_tab.py (hunk 2 above)
Comment (po09i) [reply fsleyes_plugin_shimming_toolbox/tabs/fieldmap_tab.py:143]: Same as previous comment

File: test/gui/test_tabs.py (hunk 3 above)
Comment (po09i) [reply test/gui/test_tabs.py]: Good catch

**Review by 4rnaudB — APPROVED** [review]
fsleyes version is 1.9.0, not 1.8.3 but I assume it is wanted

**Comment by po09i** [comment]
Indeed, I updated the PR description

**Comment by po09i** [comment]
Fixes https://github.com/shimming-toolbox/shimming-toolbox/issues/489

**Pull request merged**